cypress/integration/article.js

```diff
@@ -0,0 +1,124 @@
+
+const SOCIAL_URLS = [LINKEDIN_SHARE_URL, TWITTER_SHARE_URL, FACEBOOK_SHARE_URL];
+
+describe('Sample Article Page', () => {
+    it('should have a descriptive header', () => {
+        cy.visit(ARTICLE_WITH_SERIES_URL).then(() => {
+            cy.get('header').within(() => {
+                // Check title
+                cy.contains(ARTICLE_TITLE);
+                // Check pubdate
+                cy.contains('Published: Apr 01, 2020');
+                // Check tags
+                cy.checkTagListProperties();
+                // Check author
+                cy.contains('Lauren Schaefer')
+                    .should('have.prop', 'href')
+                    .and('include', '/author/lauren-schaefer');
+                // Check ATF
+                cy.get('div')
+                    .should('have.css', 'background-image')
+                    .and('include', ATF_IMAGE);
+            });
+        });
+    });
+
+    it('should have links to share content on social media', () => {
+        cy.get('[data-test="article-share-links"]').within(() => {
+            cy.get('a').each((link, index) => {
+                // Index 0 is the copy to clipboard functionality
+                // TODO: UI test clipboard link copy
+                if (index !== 0) {
+                    const url = SOCIAL_URLS[index - 1];
+                    cy.wrap(link).should('have.prop', 'href', url);
+                }
+            });
+        });
+    });
+
+    it('should have links to sub-headings in the article', () => {
+        cy.get('h2')
+            .its('length')
+            .then(numSubheadings => {
+                // Contents list should be triggered by a tooltip click
+                cy.get('[data-test="contents-list"]').should('not.exist');
+                cy.get('[data-test="contents-tooltip"]').click();
+                cy.get('[data-test="contents-list"]').within(() => {
+                    cy.get('a').should('have.length', numSubheadings);
+                    // Need to force because this lies within a tooltip
+                    cy.contains('Embrace Document Diversity').click({
+                        force: true,
+                    });
+                    cy.url().should('include', '#embrace-document-diversity');
+                });
+            });
+    });
+
+    it('should be a part of an article series', () => {
+        // Check series title
+        cy.get('[data-test="series"]').within(() => {
+            cy.contains(SERIES_TITLE);
+            cy.get('a').should('have.length', 3);
+            // Check other one goes to article
+            cy.get('a')
+                .first()
+                .should('have.prop', 'href')
+                .and('include', '/article/map-terms-concepts-sql-mongodb');
+        });
+    });
+
+    it('should have proper SEO, open graph, and twitter tags', () => {
```

Comment:
This felt like a weird Cypress test, but we had no SEO testing so I believed this was a good way to add consistency in.

Perhaps a snapshot test on the SEO would be more philosophical, but using our integration data here gives a good test as well and runs through the parser.

Reply:
I think this is reasonable. You're right, my first thought is we should test the SEO component itself, but if we're trying to confirm the page contents this makes sense as an 'article e2e test'